### PR TITLE
Die xmldom! Die!

### PR DESCRIPTION
**Pull request by JAORMX**
This PR removes go-xmldom as a dependency, finally switching everything
to use xmlquery instead. This gets rid of some parsing issues that
prevented users from seeing the whole text of instructions, descriptions
and other relevant bits of our content in the generated Kubernetes
objects.

In order to make this happen, we now had to take into account the XML
namespaces in objects.

For testing, I had to update a datastream file we were using... which
really wasn't a datastream, but a result. ARF generated by openscap
doesn't preserve the namespaces. So I updated the testing XML which
gives us more relevant results anyway.

Signed-off-by: Juan Antonio Osorio Robles <jaosorior@redhat.com>

### CODE REVIEW

**Review by jhrozek — COMMENTED** [review]
I think this is a clear improvement, but the OCIL now leaks into the instructions, see if you agree with just splitting the whitespace..

File: cmd/manager/scap_test.go [hunk 1]
```diff
@@ -26,29 +25,19 @@ var _ = Describe("Testing SCAP parsing and storage", func() {
...
 
-		It("Gets the appropriate resource URIs", func() {
+			By("parsing content for warnings")
 			expected := []string{"/apis/config.openshift.io/v1/oauths/cluster"}
 			got := getResourcePaths(contentDS, contentDS, "xccdf_org.ssgproject.content_profile_platform-moderate")
 			Expect(got).To(Equal(expected))
 		})
 	})
 
-	Context("Parsing warnings", func() {
```
[the remark this answers is not in the document]
Comment (JAORMX) [reply cmd/manager/scap_test.go:42]: We no longer parse warnings from a raw string, instead, we pass directly the `Node` object.

File: pkg/utils/parse_arf_result.go [hunk 2]
```diff
@@ -123,13 +115,13 @@ func getInstructionsForRule(rule *xmldom.Node, ocilTable nodeByIdHashTable) stri
...
 	// if not found, return empty string
-	textNode := questionNode.FindOneByName("question_text")
+	textNode := questionNode.SelectElement("ocil:question_text")
 	if textNode == nil {
 		return ""
 	}
 
 	// if found, strip the last line
-	textSlice := strings.Split(textNode.Text, "\n")
+	textSlice := strings.Split(textNode.InnerText(), "\n")
```
[the remark this answers is not in the document]
Comment (JAORMX) [reply pkg/utils/parse_arf_result.go]: thanks! Fixed.

File: cmd/manager/scap_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhrozek) [reply cmd/manager/scap_test.go:42]: thanks

**Review by jhrozek — APPROVED** [review]
/lgtm

**Pull request merged**